lib/src/runner/engine.dart

```diff
@@ -529,8 +529,8 @@ class Engine {
   Future close() async {
     _closed = true;
     if (_closedBeforeDone != null) _closedBeforeDone = true;
-    _onSuiteAddedController.close();
-    _suiteController.close();
+    await _onSuiteAddedController.close();
+    await _suiteController.close();
```

Comment:
I think these are the most likely candidates for the source of the flakiness. It's possible that there was a race condition where their events hadn't fully propagated, which would mean that the `close()` calls below didn't hit all the relevant tests.

It would be an interesting experiment to try `await`ing *only* these and seeing if that clears up the flakiness.